### ISSUE TITLE
reportGeneralTypeIssues for conditional import `Protocol`

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#2047]
```python
try:
    from typing import Protocol
except ImportError:
    from typing_extensions import Protocol


class Foo(Protocol): # ERROR: error: Argument to class must be a base class (reportGeneralTypeIssues)
    ...
```

I think it is not an error, so Pyright shouldn't report any error.

Pyright version: 1.1.148

**Comment by erictraut** [comment]
You're using exception handling as control flow and defining a new class whose base class depends on dynamic runtime behaviors. Type checkers like pyright cannot deal with dynamic behaviors like this. 

If you need to support versions of Python prior to 3.8 (when Protocol was introduced), then I recommend you unconditionally import `typing_extensions`.

If that's not feasible for some reason, you'll need to use an `if TYPE_CHECKING` conditional:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import Protocol
else:
    try:
        from typing import Protocol, TYPE_CHECKING
    except ImportError:
        from typing_extensions import Protocol
```
